### ISSUE TITLE
Bluetooth keyboards

### ISSUE THREAD

**Issue by shaunlebron** [janpaul123/paperprograms#74]
Dynamicland has bluetooth keyboards that can be pointed at programs to allow in-space editing.  Example video below:

[<img width="618" alt="screen shot 2018-08-29 at 5 41 16 pm" src="https://user-images.githubusercontent.com/116838/44819564-dfad3900-abb2-11e8-87c4-4d02de6fbc0a.png">](https://www.youtube.com/watch?v=ulkNKQQ0D9g)

I think Operating Systems always assumes one keyboard, so to allow multiple keyboards to modify programs in parallel probably requires reading bluetooth signals from each directly.

- [Web Bluetooth API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Bluetooth_API)
- [Interact with Bluetooth devices on the Web](https://developers.google.com/web/updates/2015/07/interact-with-ble-devices-on-the-web)

(Still unsure about many parts here.  Just creating a place for brainstorming)

**Comment by janpaul123** [comment]
Good idea! And yeah, we'd have to implement cursors ourselves. Might be worth finding or creating a little library that can do text editing on canvas. Also see [Bret's tweet](https://twitter.com/worrydream/status/1035951868504141826) from yesterday. 😉 

If we can't use bluetooth keyboards hooked up to one computer, another option might be having small computers for each keyboard (e.g. raspberry pi) and then having some sort of communication via the server (e.g. via websockets) to the computer that's hooked up to the projector. Something like paperprograms.org/bluetooth-keyboard/:spaceName/:keyboardProgramId?
